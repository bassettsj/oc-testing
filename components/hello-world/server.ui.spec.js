const nightmare = require('nightmare');
const oc = require('oc');
const path = require('path');
const url = require('url');

// ui
const PORT = 3030;

const BASE_URL = url.format({
  protocol: process.env.PROTOCOL || 'http',
  hostname: process.env.HOST || 'localhost',
  port: process.env.PORT || PORT
});

const SUPERMAN = 'Superman';

test(`Hello ${SUPERMAN}`, done => {
  jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
  const registry = new oc.Registry({
    baseUrl: `${BASE_URL}/`,
    local: true,
    port: PORT,
    hotReloading: false,
    path: path.resolve('./components'),
    env: { name: 'local' },
    verbosity: 0,
    discovery: true
  });

  registry.start();

  const page = nightmare({ show: true }).goto(
    `${BASE_URL}/hello-world/~preview?name=${SUPERMAN}`
  );
  page.wait(1000).end().evaluate(() => document.body.textContent).then(text => {
    expect(text).toContain(`Hello ${SUPERMAN}`);
    expect(text).toMatchSnapshot();

    registry.close();
    done();
  });
});
