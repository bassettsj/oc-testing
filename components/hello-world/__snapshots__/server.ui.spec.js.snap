// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Superman 1`] = `
"Hello Supermanwindow.oc = window.oc || {};
oc.conf = oc.conf || {};
oc.conf.templates = oc.conf.templates || [];
oc.conf.templates = oc.conf.templates.concat([{\\"type\\":\\"oc-template-jade\\",\\"version\\":\\"4.0.2\\",\\"externals\\":[{\\"global\\":\\"jade\\",\\"url\\":\\"https://cdnjs.cloudflare.com/ajax/libs/jade/1.11.0/runtime.min.js\\"}]},{\\"type\\":\\"oc-template-handlebars\\",\\"version\\":\\"4.0.1\\",\\"externals\\":[{\\"global\\":\\"Handlebars\\",\\"url\\":\\"https://cdnjs.cloudflare.com/ajax/libs/handlebars.js/4.0.6/handlebars.runtime.min.js\\"}]}]);"
`;
